python/cudf/cudf/core/udf/pipeline.py

```diff
@@ -234,8 +301,27 @@ def _kernel(retval, input_col_0, input_col_1, offset_0, offset_1, size):
     return kernel_template.format(**d)
 
 
+def _check_return_type(ty):
```

Comment:
Calling types `ty` is a very common convention in the Numba source, it might be that this is following that convention. (Though I don't have a strong opinion on what should be done in cuDF)

Reply:
I was not aware of that naming convention, thanks for the context. I tend to prefer longer names, since `ty` doesn't give me much information as a new reader. On the other hand, it does look like this could be implemented inline rather than as a separate function.

Reply:
I am fine with inlining this. However I will offer my reasoning for separating it here at least. I had sort of thought that this actually improved readability. This function exists to handle a very specific edge case and kind of needs a long comment to explain what that edge case is. It looked weird to me to insert the entire block comment into the middle of `compile_or_get`. I had hoped a reader would encounter this line of code, think "the return type is being checked, I wonder why - let me look up this function and there's the docstring". In my mind, `compile_or_get` is the top level function that implements the compilation pipeline at a high level, and it's nice to read it line by line and get the gist of what is going on without too much detail, then delve into the individual callees if needed.

Let me know if I am totally off base with this.

Reply:
I can see where you're coming from. My opinion is that large comment blocks are OK if they're necessary, and that defining a function just to _avoid_ inserting one is overkill, especially when the check is so specific to the use case where the function is being called. However, maybe we can get the best of both worlds by making the message more concise? I've given it a shot below, let me know what you think and whether that seems more suitable for inlining.